onnxruntime/core/framework/execution_steps.cc

```diff
@@ -113,5 +131,30 @@ std::string TriggerDownstreamStep::ToString() const {
   return ::onnxruntime::MakeString("TriggerDownstreamStep: trigger downstream of trigger point: ",
                                    trigger_point_index_, ".");
 }
+#ifdef ENABLE_TRAINING
+size_t GetFirstWaitOrBarrierIndexAfterPC(std::vector<std::unique_ptr<SequentialExecutionPlan::ExecutionStep>>& steps,
```

Comment:
the logic here seems not that clear to me, it seems you also handle Launch steps. let's sync offline to understand what are we doing here.

Reply:
Sure. Maybe renaming to a better name